tools/clang/unittests/HLSL/DxilContainerTest.cpp

```diff
@@ -1154,6 +1155,60 @@ TEST_F(DxilContainerTest, CompileGS_CheckPSV0_ViewID) {
   }
 }
 
+TEST_F(DxilContainerTest, Compile_CheckPSV0_EntryFunctionName) {
+  if (m_ver.SkipDxilVersion(1, 8))
+    return;
+
+  auto CheckEntryFunctionName = [&](const char *EntryPointName,
+                                    const char *Source, const char *Profile) {
+    CComPtr<IDxcBlobEncoding> pSource;
+    CComPtr<IDxcBlob> pProgram;
+    CComPtr<IDxcOperationResult> pResult;
+    CreateBlobFromText(Source, &pSource);
+    CComPtr<IDxcCompiler> pCompiler;
+    VERIFY_SUCCEEDED(CreateCompiler(&pCompiler));
+    CA2W wEntryPointName(EntryPointName);
+    CA2W wProfile(Profile);
+    VERIFY_SUCCEEDED(pCompiler->Compile(pSource, L"hlsl.hlsl", wEntryPointName,
+                                        wProfile, nullptr, 0, nullptr, 0,
+                                        nullptr, &pResult));
+    HRESULT hrStatus;
+    VERIFY_SUCCEEDED(pResult->GetStatus(&hrStatus));
+    VERIFY_SUCCEEDED(hrStatus);
+    VERIFY_SUCCEEDED(pResult->GetResult(&pProgram));
+
+    // Get PSV0 part
+    CComPtr<IDxcContainerReflection> containerReflection;
+    VERIFY_SUCCEEDED(m_dllSupport.CreateInstance(CLSID_DxcContainerReflection,
+                                                 &containerReflection));
+    VERIFY_SUCCEEDED(containerReflection->Load(pProgram));
+    uint32_t partIdx = 0;
+    VERIFY_SUCCEEDED(containerReflection->FindFirstPartKind(
+        (uint32_t)hlsl::DxilFourCC::DFCC_PipelineStateValidation, &partIdx));
+    CComPtr<IDxcBlob> pBlob;
+    VERIFY_SUCCEEDED(containerReflection->GetPartContent(partIdx, &pBlob));
+
+    // Look up EntryFunctionName in PSV0
+    DxilPipelineStateValidation PSV;
+    PSV.InitFromPSV0(pBlob->GetBufferPointer(), pBlob->GetBufferSize());
+
+    // We should have a valid PSVRuntimeInfo3 structure
+    PSVRuntimeInfo3 *pInfo3 = PSV.GetPSVRuntimeInfo3();
+    VERIFY_IS_NOT_NULL(pInfo3);
+
+    // Finally, verify that the entry function name in PSV0 matches
+    VERIFY_ARE_EQUAL_STR(EntryPointName, PSV.GetEntryFunctionName());
+  };
+
+  const char src_PSMain[] = "float MyPSMain() : SV_Target { return 0; }";
+  const char src_VSMain[] = "float4 MyVSMain() : SV_Position { return 0; }";
+  const char src_CSMain[] = "[numthreads(1,1,1)] void MyCSMain() {}";
+
+  CheckEntryFunctionName("MyPSMain", src_PSMain, "ps_6_0");
+  CheckEntryFunctionName("MyVSMain", src_VSMain, "vs_6_0");
+  CheckEntryFunctionName("MyCSMain", src_CSMain, "cs_6_0");
```

Comment:
Would it be possible to test for library shaders where no entry point exists? Would it be impossible to get a PSV3 object from a library? Maybe we could test that like this in the library case:
```
PSVRuntimeInfo3 *pInfo3 = PSV.GetPSVRuntimeInfo3();
    VERIFY_IS_NULL(pInfo3); // IS NULL
```

Reply:
`PSV0` is only used for non-library shaders.  For libraries, `RDAT` replaces `PSV0` for supplying runtime data.

Also, `PSV3` isn't a thing.  That `0` in `PSV0` is a higher level versioning for the part protocol itself, in case we needed to make a breaking change to that protocol.  There's no need to version that at this point.

Reply:
I guess `PSV3` was just shorthand for `PSVRuntimeInfo3` though...

Reply:
The test case is not relevant, because there's no `PSV0` part to load into PSV in the first place.  This change doesn't affect any of that, and the validator would fail if you had a library with an unexpected `PSV0` part.

Reply:
Ok, that resolves my concern